mapeamento/mapeadores/spiders/bases/mapeador_semantico.py

```diff
@@ -0,0 +1,167 @@
+            combinations.update(self.progressive_colapsed_words_set(name))
+        
+        combinations.discard("")
+        return combinations
+
+    def path_generator(self, city):
+        """
+        same as domain_generator(), but special characters are allowed
+        """
+        combinations = self.domain_generator(city)
+
+        for name in self.add_prefeitura_to_name(city):
+            combinations.add(self.name_with_underline(name))    
+            combinations.add(self.name_with_hifen(name))        
+
+        return combinations
+
+    def add_prefeitura_to_name(self, name):
+        return [
+            name,
+            f"prefeitura {name}",
+            f"prefeitura de {name}",
+            f"prefeitura municipal {name}",
+            f"prefeitura municipal de {name}",
+        ]
+    
+    def name_no_blankspaces(self, name): 
+        return slugify(name, separator="", replacements=self.replacements)
+
+    def name_with_underline(self, name): 
+        return slugify(name, separator="_", replacements=self.replacements)
+
+    def name_with_hifen(self, name): 
+        return slugify(name, separator="-", replacements=self.replacements)
+
+    def name_parts_set(self, name): 
+        stopwords = self.stopwords + ["prefeitura", "municipal"]
+        return set(slugify(name, separator=" ", replacements=self.replacements, stopwords=stopwords).split())
+
+    def progressive_colapsed_words_set(self, name):
+        """
+        Iterates over a words list creating combinations following
+        abbreviation + rest pattern. 
+        Stopwords are allowed to exist in "rest" part, but not 
+        in "abbreviation" part.
+        
+        Example: 
+        input: Prefeitura Municipal Santo Antonio do Paraiso
+        outputs:
+        - P Municipal Santo Antonio do Paraiso
+        - P M Santo Antonio do Paraiso
+        - P M S Antonio do Paraiso
+        - P M S A do Paraiso
+        - P M S A Paraiso
+        - P M S A P
+        """
+        colapsed_words = set()
+        words = slugify(name, separator=" ", replacements=self.replacements).split()
+
+        abbrev = ""
+        for i in range(len(words)):
+            word = words[i]
+            
+            if word not in self.stopwords:
+                abbrev += word[0]
+                rest = words[i+1:]                
+                colapsed_words.add(self.join_parts(abbrev, rest))
+                colapsed_words.add(self.join_parts(abbrev, self._wo_stopwords(rest)))
+        
+        return colapsed_words
+
+    def _wo_stopwords(self, sublist):
+        return [x for x in sublist if x not in self.stopwords]
+    
+    def join_parts(self, inicial, sublist):
+        """
+        Concatenate text parts, discarding one letter only possibility
+        For example, prevents returning "r" for "Recife"
+        """
+        opt = f"{inicial}{''.join(sublist)}"
+        if len(opt) > 1:
+            return opt
+        return ""
```

Comment:
Para não criar dois métodos separados (1. concatenação e 2. descarte pelo tamanho), que também poderia ser uma solução válida, deixaria o nome da função menos genérico e colocaria algo como `valid` no nome para criar a expectativa de que o retorno tem alguma condição e o método não faz apenas concatenar strings
```suggestion
    def join_valid_collapsed_variations(self, initial, sublist):
        """
        Concatenate text parts, discarding one letter only possibility
        For example, prevents returning "r" for "Recife"
        """
        collapsed = f"{initial}{''.join(sublist)}"
        if len(opt) > 1:
            return collapsed
        return ""
```

Reply:
como falamos sobre permitir todas as combinações inicialmente e só ao fim descartar os casos irrelevantes todos de uma vez, acabou de esse método não precisou mais existir. A única linha relevante dele 
> collapsed = f"{initial}{''.join(sublist)}"

foi adicionada em `progressive_collapsed` mesmo. O descarte de tamanho pequeno (linhas 165 e 166 acima) é feito por outro método agora